server/src/buffer.rs

```diff
@@ -79,13 +81,25 @@ impl Buffer {
             };
 
             if oldest_is_persisted {
-                self.remove_oldest_segment();
+                let _ = self.remove_oldest_segment();
```

Comment:
```suggestion
                self.remove_oldest_segment();
```

I don't think the `let _` is needed

Reply:
For some reason, I anticipated clippy yelling at me for not assigning the return value :smile: